src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -112,6 +112,22 @@ void gdb_value_extractort::analyze_symbols(
     CHECK_RETURN(value.pointee.empty() || (value.pointee == id));
 
     values.insert({value.address, symbol_expr});
+
+    const symbolt &symbol = ns.lookup(id);
+    const symbol_exprt actual_expr = symbol.symbol_expr();
+    if(actual_expr.type().id() != ID_pointer)
+    {
+      memory_map[id2string(id)] = value;
```

Comment:
I'm not entirely sure this should be indexed by `std::string`, wouldn't `irep_idt` be preferable?

Reply:
Works with `irep_idt` as well. Changed.